runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobInvoker.java

```diff
@@ -66,7 +66,9 @@ protected JobInvocation invokeWithExecutor(
         String.format("%s_%s", flinkOptions.getJobName(), UUID.randomUUID().toString());
 
     if (FlinkPipelineOptions.AUTO.equals(flinkOptions.getFlinkMaster())) {
-      flinkOptions.setFlinkMaster(serverConfig.getFlinkMaster());
+      // Although Flink uses Rest, it expects the address not to contain a http scheme
+      String flinkMaster = serverConfig.getFlinkMaster().trim().replaceFirst("^http://", "");
```

Comment:
Not sure whether we should allow `https`, since it is not supported anyway. I'm ok with adding support, if we figure out a way for Python to support it.